File: core/src/main/java/com/ustadmobile/core/view/SelectMultipleTreeDialogView.java
package com.ustadmobile.core.view;


import com.ustadmobile.lib.db.entities.Location;

import java.util.List;

/**
 * SelectMultipleTreeDialog Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
public interface SelectMultipleTreeDialogView extends UstadView {

    String VIEW_NAME = "SelectMultipleTreeDialog";

    String ARG_LOCATIONS_SET = "LocationsSelected";

    void populateTopLocation(List<Location> locations);

    void setTitle(String title);

    /**
     * For Android: closes the activity.
     */
    void finish();

}

File: core/src/commonMain/kotlin/com/ustadmobile/staging/core/view/SelectMultipleTreeDialogView.kt
package com.ustadmobile.core.view


import com.ustadmobile.lib.db.entities.Location

/**
 * SelectMultipleTreeDialog Core View extends Core UstadView. Will be implemented
 * on various implementations.
 */
interface SelectMultipleTreeDialogView : UstadView {

    fun populateTopLocation(locations: List<Location>)

    /**
     * For Android: closes the activity.
     */
    fun finish()

    companion object {

        val VIEW_NAME = "SelectMultipleTreeDialog"
    }

}
